### PR TITLE
added state handling for when the channel is closed.  

### PR DESCRIPTION
**Pull request by lamchakchan**
## Description
- Updated documentation on limitation of `config` setter in reflection with when the connection is affected.
- Cleared out `SubscriptionManager` when `channel` is closed.

## Motivation and Context
Cleaning up low level edge cases.

## Types of changes
- [ ] Documentation only (no changes to either `lib/` or `test/` files)
- [x] Bug fix (non-breaking change which fixes an issue. you didn't modify existing tests)
- [ ] New feature (non-breaking change which adds functionality. you added at least one new test)
- [ ] Breaking change (fix or feature that would cause existing functionality to change. you had to modify existing tests.)

## Checklist:
- [x] I have read the [**CONTRIBUTING**](https://github.com/xogroup/bunnybus/blob/master/.github/CONTRIBUTING.md) document.
- [x] My code follows the [Hapi.js style guide](https://github.com/hapijs/contrib/blob/master/Style.md).
- [x] I have updated the documentation as needed.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -86,6 +86,8 @@ Setter and Getter for singleton configuration. Accepts the following optional pr
  * `globalExchange` - value of the exchange to transact through for message publishing.  This is the default used when one is not provided as an within the `options` for any `BunnyBus` methods that supports one transactionally.  Defaults to `default-exchange`. *[string]* **Optional**
  * `prefetch` - value of the maximum number of unacknowledged messages allowable in a channel.  Defaults to `5`. *[number]* **Optional**
 
+Note that updates in the options directed at changing connection string will not take affect immediately.  `_closeConnection()`](#_closeConnectioncallback) needs to be called manually to invoke a new connection with new settings.
```
Comment (WesTyler) [inline API.md]: Typo in the `closeConnection` anchor link - you're miss the open square bracket

[`_closeConnection()`]

File: lib/states/subscriptionManager.js [hunk 2]
```diff
@@ -99,6 +99,13 @@ class SubscriptionManager extends EventEmitter {
         return false;
     }
 
+    clearAll() {
```
Comment (WesTyler) [inline lib/states/subscriptionManager.js]: I like this approach :D
